sos/report/plugins/foreman_installer.py

```diff
@@ -0,0 +1,87 @@
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import (Plugin, RedHatPlugin, DebianPlugin,
+                                UbuntuPlugin)
+
+
+class ForemanInstaller(Plugin, DebianPlugin, UbuntuPlugin):
+
+    short_desc = 'Foreman installer and maintainer'
+
+    plugin_name = 'foreman_installer'
+    profiles = ('sysmgmt',)
+    packages = ('foreman-installer', 'rubygem-foreman_maintain')
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/foreman-installer/",
+            "/var/log/foreman-installer/*",
+            "/var/log/foreman-maintain/*",
+        ])
+
+        self.add_cmd_output([
+            'foreman-maintain service status',
+        ])
+
+    def postproc(self):
+        install_logs = "/var/log/foreman-installer/"
+        logsreg = r"((foreman.*)?(\"::(foreman(.*?)|katello).*)?((::(.*)::.*" \
+                  r"(passw|cred|token|secret|key).*(\")?:)|(storepass )" \
+                  r"|(password =)))(.*)"
+        self.do_path_regex_sub(install_logs, logsreg, r"\1 ********")
+        # need to do two passes here, debug output has different formatting
+        logs_debug_reg = (r"(\s)+(Found key: (\"(foreman(.*?)|katello)"
+                          r"::(.*(token|secret|key|passw).*)\") value:) "
+                          r"(.*)")
+        self.do_path_regex_sub(install_logs, logs_debug_reg, r"\1 \2 ********")
+        # also hide passwords in yet different formats
+        self.do_path_regex_sub(
+            install_logs,
+            r"(\.|_|-)password(=\'|=|\", \")(\w*)",
+            r"\1password\2********")
+        self.do_path_regex_sub(
+            "/var/log/foreman-installer/foreman-proxy*",
+            r"(\s*proxy_password\s=) (.*)",
+            r"\1 ********")
+        self.do_path_regex_sub(
+            "/var/log/foreman-maintain/foreman-maintain.log*",
+            r"(((passw|cred|token|secret)=)|(password ))(.*)",
+            r"\1********")
+        # all scrubbing applied to configs must be applied to installer logs
+        # as well, since logs contain diff of configs
+        self.do_path_regex_sub(
+            r"(/etc/foreman-(installer|maintain)/(.*)((conf)(.*)?))|(%s)"
+            % install_logs,
+            r"((\:|\s*)(passw|cred|token|secret|key).*(\:\s|=))(.*)",
+            r"\1********")
+        # yaml values should be alphanumeric
+        self.do_path_regex_sub(
+            r"(/etc/foreman-(installer|maintain)/(.*)((yaml|yml)(.*)?))|(%s)"
+            % install_logs,
+            r"((\:|\s*)(passw|cred|token|secret|key).*(\:\s|=))(.*)",
+            r'\1"********"')
+
+
+# Add Red Hat Insights tags for RedHatPlugin only
+
+class RedHatForemanInstaller(ForemanInstaller, RedHatPlugin):
+
+    def setup(self):
+
+        self.add_file_tags({
+            '/var/log/foreman-installer/satellite.log.*':
```

Comment:
Is this a glob or regex? If it's a regex, then it's ok and will match `/var/log/foreman-installer/satellite.log` itself.

Reply:
Yes, I do see that tag being created.

Reply:
I do wonder if we should also do the same for `capsule.log`.

Reply:
Afaik Insights does not demand/use it, but we can enhance the tags for future (and also to refer to it as `capsule_installer_log`).